### PR TITLE
Added Configuring Applications and Components as Code with Kustomize

### PR DESCRIPTION
**Pull request by gtrivedi88**
Purpose: The purpose of this PR is to develop the releasing an application content. I have copied the AppStudio content directly. We have to review and update it.

### CODE REVIEW

File: docs/modules/ROOT/pages/how-tos/configuring/configuration-as-code.adoc [hunk 1]
```diff
@@ -1,4 +1,322 @@
...
+  targetPort: 8080
+  source:
+      git:
+        url: gitUrl
+        context: ./
+        dockerfileUrl: ContainerFileLocation
+        revision: defaultBranch
+----
+
+NOTE: This will create a component with a custom build pipeline.  If you wish to use the default build, omit the `build.appstudio.openshift.io/request: configure-pac` annotation.
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/configuring/configuration-as-code.adoc:81]: I don't think we need the second sentence. Is the first consistent with other pages?

File: docs/modules/ROOT/pages/how-tos/configuring/configuration-as-code.adoc [hunk 2]
```diff
@@ -1,4 +1,322 @@
...
+    path: component-patch.yaml
+  - target:
+      kind: Component
+      name: component-b
+    path: exception-component-patch-1.yaml
+----
+
+NOTE: This kustomizaton applies a suffix to all resources, we recommend doing this to ensure uniqueness and make it easier to identify components and applications.  
+
+NOTE: You can define more than one exceptional patch and match component names through regex.
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/configuring/configuration-as-code.adoc:235]: Are these NOTEs using the proper formats?

File: docs/modules/ROOT/pages/how-tos/configuring/configuration-as-code.adoc [hunk 3]
```diff
@@ -1,4 +1,322 @@
...
+│           ├── exception-component-patch-1.yaml
+│           └── kustomization.yaml
+└── README.md
+----
+
+== Defining Multiple Applications
+
+If you wish to define multiple applications with different sets of components and versions for each application, you can replicate the configuration overviewed above for `application-a` for a second application and include it as an additional application base and overlays in the `overlay` directory. 
+
+This is the preferred way to define multiple applications within an application category (ex. Operators) or family/product organization (ex. Red Hat Advanced Cluster Management) as code because it allows you to make bulk configurations to your base component yaml (such as enabling muti-arch or labelling/ownership) in a single place, the component base, rather than multiple places.
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/configuring/configuration-as-code.adoc:275]: We shouldn't call out a specific Red Hat product.

**Review by arewm — CHANGES_REQUESTED** [review]
General question: Should Component and Application be capitalized throughout?

**Comment by ifireball** [comment]
> @ifireball , it feels like there is some overlap between this documentation and the project version controller. Would you mind reviewing this PR?

Indeed it does seem like at least the multi-version section here could be simplified by using the project version controller. Maybe others as well. @gtrivedi88 can you please have a look at the [multi-version HOWTO](https://redhat-appstudio.github.io/docs.appstudio.io/Documentation/main/how-to-guides/proc_multiversion/) and see if the information there helps you simplify this HOWTO?

File: docs/modules/ROOT/pages/how-tos/configuring/configuration-as-code.adoc (hunk 2 above)
Comment (gtrivedi88) [reply docs/modules/ROOT/pages/how-tos/configuring/configuration-as-code.adoc]: Yes

**Comment by gtrivedi88** [comment]
As per discussion in slack, added a note at the end: https://redhat-internal.slack.com/archives/C0715J0KEBS/p1715577674171459

File: docs/modules/ROOT/pages/how-tos/configuring/configuration-as-code.adoc [hunk 4]
```diff
@@ -1,4 +1,351 @@
...
+│           ├── application-patch.yaml
+│           ├── component-patch.yaml
+│           ├── exception-component-patch-1.yaml
+│           └── kustomization.yaml
+└── README.md
+----
+
+NOTE: You can also modify the project structure to fit your own needs by moving application bases and component definitions to different levels, but this configuration offers the most layered encapsulation across applications and application versions.
+
+IMPORTANT: Managing multiple related components and applications can be challenging. Refer https://redhat-appstudio.github.io/docs.appstudio.io/Documentation/main/how-to-guides/proc_multiversion/ [Managing multiple software versions] to manage multiple related components and applications.
```
[the remark this answers is not in the document]
Comment (gtrivedi88) [reply docs/modules/ROOT/pages/how-tos/configuring/configuration-as-code.adoc:353]: Yes

**Comment by gtrivedi88** [comment]
@arewm All comments resolved. Could you please approve this PR or let me know if you have any additional suggestions.

**Review by arewm — APPROVED** [review]

**Pull request merged**